src/v2/Apps/Algolia/routes/AlgoliaResults/Components/__tests__/AlgoliaResultsList.jest.tsx

```diff
@@ -0,0 +1,161 @@
+} from "v2/__generated__/AlgoliaResultsList_Test_Query.graphql"
+import {
+  AlgoliaResultsListProps,
+  AlgoliaResultsListFragmentContainer,
+} from "../AlgoliaResultsList"
+import { SearchResults } from "react-instantsearch-core"
+import { AlgoliaHit } from "v2/Apps/Algolia/types"
+
+jest.unmock("react-relay")
+jest.mock("v2/System/Router/useRouter", () => ({
+  useRouter: () => ({
+    match: {
+      location: {
+        query: {
+          query: "query",
+        },
+      },
+    },
+  }),
+  useIsRouteActive: () => false,
+}))
+
+const onClickMock = jest.fn()
+
+describe("AlgoliaResultsList", () => {
+  const getWrapper = (
+    props?: Partial<Omit<AlgoliaResultsListProps, "algolia">>
+  ) => {
+    return setupTestWrapperTL<AlgoliaResultsList_Test_Query>({
+      Component: ({ system }) => {
+        if (system?.algolia) {
+          return (
+            <AlgoliaResultsListFragmentContainer
+              algolia={system.algolia}
+              searchState={{}}
+              searchResults={SearchResultsFixture}
+              searching={false}
+              isSearchStalled={false}
+              searchingForFacetValues={[]}
+              allSearchResults={SearchResultsFixture as any}
+              error={null as any}
+              {...props}
+            />
+          )
+        }
+
+        return null
+      },
+      query: graphql`
+        query AlgoliaResultsList_Test_Query @raw_response_type {
+          system {
+            algolia {
+              ...AlgoliaResultsList_algolia
+            }
+          }
+        }
+      `,
+    })
+  }
+
+  beforeEach(() => {
+    onClickMock.mockClear()
+  })
+
+  it("should render hits", () => {
+    const { renderWithRelay } = getWrapper()
+
+    renderWithRelay({
+      Algolia: () => ({
+        ...AlgoliaIndicesFixture,
+      }),
+    })
+
+    expect(screen.getByText("Hit Name One")).toBeInTheDocument()
+    expect(screen.getByText("Hit Name Two")).toBeInTheDocument()
+    expect(screen.getByText("Hit Name Three")).toBeInTheDocument()
+  })
+
+  it("should render empty message", () => {
+    const { renderWithRelay } = getWrapper({
+      searchResults: {
+        ...SearchResultsFixture,
+        nbHits: 0,
+        hits: [],
+      },
+    })
+
+    renderWithRelay({
+      Algolia: () => ({
+        ...AlgoliaIndicesFixture,
+      }),
+    })
+
+    const title = `Sorry, we couldn’t find an Article for “ABC”`
+    const message = "Please try searching again with a different spelling."
+
+    expect(screen.getByText(title)).toBeInTheDocument()
+    expect(screen.getByText(message)).toBeInTheDocument()
+  })
+})
+
+const AlgoliaIndicesFixture: NonNullable<
+  AlgoliaResultsList_Test_QueryRawResponse["system"]
+>["algolia"] = {
+  indices: [
+    {
+      displayName: "Article",
+      name: "article",
+    },
+    {
+      displayName: "Artist",
+      name: "artist",
+    },
+  ],
+}
+
+const SearchResultsFixture: SearchResults<AlgoliaHit> = {
```

Comment:
We should avoid fixtures if possible as they become a burden to maintain in the future. That's not to say that we should never use them, only that if we do, we should only supply fixture data for the props that we're asserting against in the test, or that which is required in order to get the component to render. (Relay test tools auto-fixture everything else). If these conditions are required here then ignore this comment!

Reply:
> If these conditions are required here then ignore this comment!

Yes, all these fixtures are necessary here in order for the component to render without errors and then to cover it with tests correctly